code/modules/mob/living/living_movement.dm

```diff
@@ -210,6 +210,10 @@ default behaviour is:
 /mob/living/Moved(var/atom/oldloc, direct, forced, movetime)
 	. = ..()
 	handle_footstep(loc)
+	if(is_shifted)
+		is_shifted = FALSE
+		pixel_x = default_pixel_x
+		pixel_y = default_pixel_y
```

Comment:
By tracking custom offsets independently, we just subtract pixel_x/y by those values and set the tracking vars to 0
```suggestion
if(custom_pixel_x)
	pixel_x -= custom_pixel_x
	custom_pixel_x = 0

if(custom_pixel_y)
	pixel_y -= custom_pixel_y
	custom_pixel_y = 0
```